contracts/ModifiedTokenDistro.sol

```diff
@@ -0,0 +1,359 @@
+     */
+    function globallyClaimableAt(uint256 timestamp) public view override returns (uint256) {
+        if (timestamp < startTime) return 0;
+        if (timestamp < cliffTime) return initialAmount;
+        if (timestamp > startTime + duration) return totalTokens;
+
+        uint256 deltaTime = timestamp - startTime;
+        return initialAmount + (deltaTime * lockedAmount) / duration;
+    }
+
+    /**
+     * Function to get the unlocked tokes at some moment for a specific address
+     * @param recipient account to query
+     * @param timestamp Instant of time in which the calculation is made
+     */
+    function claimableAt(address recipient, uint256 timestamp) public view override returns (uint256) {
+        require(!hasRole(DISTRIBUTOR_ROLE, recipient), 'TokenDistro::claimableAt: DISTRIBUTOR_ROLE_CANNOT_CLAIM');
+        require(timestamp >= getTimestamp(), 'TokenDistro::claimableAt: NOT_VALID_PAST_TIMESTAMP');
+        uint256 unlockedAmount = (globallyClaimableAt(timestamp) * balances[recipient].allocatedTokens) / totalTokens;
+
+        return unlockedAmount - balances[recipient].claimed;
+    }
+
+    /**
+     * Function to get the unlocked tokens for a specific address. It uses the current timestamp
+     * @param recipient account to query
+     */
+    function claimableNow(address recipient) public view override returns (uint256) {
+        return claimableAt(recipient, getTimestamp());
+    }
+
+    /**
+     * Function that allows the DEFAULT_ADMIN_ROLE to change a recipient in case it wants to cancel an allocation
+     * @dev The change can only be made when cancelable is true and to an address that has not previously received
+     * an allocation and the distributor cannot change its address
+     *
+     * Emits a {ChangeAddress} event.
+     *
+     */
+    function transferAllocation(address prevRecipient, address newRecipient) external override {
+        require(
+            balances[prevRecipient].allocatedTokens > 0, 'TokenDistro::transferAllocation: NO_ALLOCATION_TO_TRANSFER'
+        );
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), 'TokenDistro::transferAllocation: ONLY_ADMIN_ROLE');
+
+        // require(
+        //     balances[newRecipient].allocatedTokens == 0 &&
+        //         balances[newRecipient].claimed == 0,
+        //     "TokenDistro::cancelAllocation: ADDRESS_ALREADY_IN_USE"
+        // );
+
+        require(
+            !hasRole(DISTRIBUTOR_ROLE, prevRecipient) && !hasRole(DISTRIBUTOR_ROLE, newRecipient),
+            'TokenDistro::transferAllocation: DISTRIBUTOR_ROLE_NOT_A_VALID_ADDRESS'
+        );
+        // balance adds instead of overwrites
+        balances[newRecipient].allocatedTokens =
+            balances[prevRecipient].allocatedTokens.add(balances[newRecipient].allocatedTokens);
+        balances[prevRecipient].allocatedTokens = 0;
+
+        balances[newRecipient].claimed = balances[prevRecipient].claimed.add(balances[newRecipient].claimed);
```

Comment:
The arithmetic can be done without use of SafeMath, like what has happened in `changeAddress` function. Plus, I think since these two methods doing similar logic, let's define an internal method which transfers balance of an account to another and use that inside these two.